temba/contacts/tasks.py

```diff
@@ -117,3 +119,66 @@ def check_elasticsearch_lag():
                 return False
 
     return True
+
+
+@nonoverlapping_task(track_started=True, name="upload_channels_and_flows_to_es_contacts")
+def upload_channels_and_flows_to_es_contacts():
+    raw_query = """
+    SELECT id,
+    (
+        SELECT array_to_json(array_agg(row_to_json(u)))
+        FROM (
+            SELECT scheme,
+                path,
+                channels_channel.uuid as channel_uuid,
+                channels_channel.name as channel_name
+            FROM contacts_contacturn
+            LEFT JOIN channels_channel ON contacts_contacturn.channel_id = channels_channel.id
+            WHERE contact_id = contacts_contact.id
+        ) u
+    ) as urns_channels,
+    (
+        SELECT array_to_json(array_agg(row_to_json(f)))
+        FROM (
+            SELECT DISTINCT ff.uuid, ff.name
+            FROM flows_flowrun fr
+            INNER JOIN flows_flow ff on ff.id = fr.flow_id
+            WHERE fr.contact_id = contacts_contact.id
+        ) f
+    ) as flows
+    FROM contacts_contact
+    WHERE contacts_contact.is_active = true 
+        AND contacts_contact.org_id = %d 
+        AND contacts_contact.id > %d
+        AND (
+            SELECT greatest(max(flows_flowrun.created_on), contacts_contact.modified_on)
+            FROM flows_flowrun WHERE flows_flowrun.contact_id = contacts_contact.id
+        ) >= '%s'
+    ORDER BY contacts_contact.id
+    LIMIT 500000;
```

Comment:
@ihor-palii - do you think there's the possibility to have more than 500k on this query's return?

Reply:
The same limit was on the rp-indexer, probably for performance reasons (not sure whether for DB or Elasticsearch). I think that one org could contain more than 500K contacts. That is why I added that limit here. Just in case if org contains more than 500K contacts, I split them into chunks.